### ISSUE TITLE
Price fetching needs to adapt to respective asset chainId

### ISSUE THREAD

**Issue by kremalicious** [oceanprotocol/market#658]
Right now no price display works when user is not connected to same network as the asset which is especially noticeable on frontpage. This needs to be refactored so that price fetching is always based on some passed `chainId` most likely from the DDO.

This pretty much requires some way to switch Apollo client endpoints based on `ddo.chainId`

**Comment by kremalicious** [comment]
Just some research notes: Apollo by design always expects one client at a time so we need to workaround it for which some solutions are:

- https://medium.com/open-graphql/apollo-multiple-clients-with-react-b34b571210a5
- https://www.loudnoises.us/next-js-two-apollo-clients-two-graphql-data-sources-the-easy-way/

I think the latter should be preferred, as we then end up in components with:

```js
const { data, error, loading } = useQuery(GET_STUFF, {
  context: { clientName: "rinkeby" }
});
```

where we could also just pass `subgraphUri` most likely:

```js
const { data, error, loading } = useQuery(GET_STUFF, {
  context: { subgraphUri }
});
```

This is generally required as we move into a UI where on one screen we have to get data from multiple subgraphs at the same time.

**Comment by mihaisc** [comment]
Maybe this is a good opportunity to explore this https://github.com/oceanprotocol/market/issues/608. Lately we have only been working around Apollo

**Comment by bogdanfazakas** [comment]
closed by #693
